src/main/java/net/neoforged/neoforge/network/handlers/ClientPayloadHandler.java

```diff
@@ -136,4 +138,16 @@ private static <T extends AbstractContainerMenu> void createMenuScreen(Component
             mc.setScreen(s);
         });
     }
+
+    public void handle(AuxiliaryLightDataPayload msg, PlayPayloadContext context) {
+        context.workHandler().execute(() -> {
```

Comment:
Use a `submit`  with proper exception handling, and disconnect the client when an error here occurs.

I do not think that connecting a client to part of the world were de-serialization failed is a good idea.